src/main/scala/shine/DPIA/FunctionalPrimitives/ForeignFunction.scala

```diff
@@ -34,23 +34,7 @@ final case class ForeignFunction(funDecl: ForeignFunction.Declaration,
 
   override def acceptorTranslation(A: Phrase[AccType])
                                   (implicit context: TranslationContext): Phrase[CommType] = {
-    import TranslationToImperative._
-
-    def recurse(ts: Seq[(Phrase[ExpType], DataType)],
-                exps: Seq[Phrase[ExpType]],
-                inTs: Seq[DataType]): Phrase[CommType] = {
-      ts match {
-        // with only one argument left to process return the assignment of the function call
-        case Seq((arg, inT)) =>
-          con(arg)(λ(expT(inT, read))(e =>
-            A :=| outT | ForeignFunction(funDecl, inTs :+ inT, outT, exps :+ e)))
-        // with a `tail` of arguments left, recurse
-        case Seq((arg, inT), tail@_*) =>
-          con(arg)(λ(expT(inT, read))(e => recurse(tail, exps :+ e, inTs :+ inT)))
-      }
-    }
-
-    recurse(args zip inTs, Seq(), Seq())
+    ???
```

Comment:
Can you explain why it is required (or a good idea) to remove the acceptor translation?

Reply:
the `ForeignFunction` primitive is `read`, and so should only trigger continuation translation I believe.

Reply:
I may be wrong on this, but doesn't a top-level `read` primitive that returns a simple type like `f32` still trigger the acceptor translation?

I think I am facing exactly this in my current implementation of dependent pair match, which in one example is of type `exp[f32, read]`, and triggers the acceptor translation?

Again, I am rusty in `shine`, but it's worth a check